### PR TITLE
Azure Managed Services RP .net SDK

### PR DESCRIPTION
**Pull request by rajivnandivada**
Swagger used: Azure/azure-rest-api-specs#5328 (still in PR)

### CODE REVIEW

File: src/SDKs/ManagedServices/Management.ManagedServices/Microsoft.Azure.Management.ManagedServices.csproj [hunk 1]
```diff
@@ -1,22 +1,26 @@
...
-	<PropertyGroup>
-		<PackageId>Microsoft.Azure.Management.ManagedServices</PackageId>
-		<Description>Microsoft Azure Management ManagedServices library</Description>
-		<AssemblyName>Microsoft.Azure.Management.ManagedServices</AssemblyName>
-		<Version>0.9.0-preview</Version>
-		<PackageTags>Azure ManagedServices;</PackageTags>
+  <PropertyGroup>
+    <PackageId>Microsoft.Azure.Management.ManagedServices</PackageId>
+    <Description>Microsoft Azure Managed Services Management Library</Description>
+    <Version>0.9.0</Version>
```
Comment (dsgouda) [inline src/SDKs/ManagedServices/Management.ManagedServices/Microsoft.Azure.Management.ManagedServices.csproj]: Are you ready to publish a stable nuget version? If yes, the version must be bumped to 1.0.0

File: src/SDKs/ManagedServices/Management.ManagedServices/Microsoft.Azure.Management.ManagedServices.csproj [hunk 2]
```diff
@@ -1,22 +1,26 @@
...
+    <AssemblyName>Microsoft.Azure.Management.ManagedServices</AssemblyName>
+    <PackageTags>Microsoft Azure Managed Services Management;Managed Services;Managed Services Management;</PackageTags>
     <PackageReleaseNotes>
+      <![CDATA[
+
```
Comment (dsgouda) [inline src/SDKs/ManagedServices/Management.ManagedServices/Microsoft.Azure.Management.ManagedServices.csproj]: Please add the `PackageReleaseNotes`

File: src/SDKs/ManagedServices/Management.ManagedServices/Microsoft.Azure.Management.ManagedServices.csproj [hunk 3]
```diff
@@ -1,22 +1,26 @@
...
+    <AssemblyName>Microsoft.Azure.Management.ManagedServices</AssemblyName>
+    <PackageTags>Microsoft Azure Managed Services Management;Managed Services;Managed Services Management;</PackageTags>
     <PackageReleaseNotes>
+      <![CDATA[
+      
+      ]]>
     </PackageReleaseNotes>
-	</PropertyGroup>
-	<PropertyGroup>
-		<TargetFrameworks>$(SdkTargetFx)</TargetFrameworks>
```
Comment (dsgouda) [inline src/SDKs/ManagedServices/Management.ManagedServices/Microsoft.Azure.Management.ManagedServices.csproj]: Any reason why this was removed?

File: src/SDKs/_metadata/managedservices_resource-manager.txt [hunk 4]
```diff
@@ -0,0 +1,14 @@
+﻿Installing AutoRest version: latest
+AutoRest installed successfully.
+Commencing code generation
+Generating CSharp code
```
Comment (dsgouda) [inline src/SDKs/_metadata/managedservices_resource-manager.txt]: Please delete this file and regenerate the code, commit the new file.

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: src/SDKs/ManagedServices/Management.ManagedServices/Microsoft.Azure.Management.ManagedServices.csproj (hunk 1 above)
Comment (rajivnandivada) [reply src/SDKs/ManagedServices/Management.ManagedServices/Microsoft.Azure.Management.ManagedServices.csproj]: That makes sense. I will bump it.

File: src/SDKs/ManagedServices/Management.ManagedServices/Microsoft.Azure.Management.ManagedServices.csproj (hunk 2 above)
Comment (rajivnandivada) [reply src/SDKs/ManagedServices/Management.ManagedServices/Microsoft.Azure.Management.ManagedServices.csproj]: will do thanks

File: src/SDKs/ManagedServices/Management.ManagedServices/Microsoft.Azure.Management.ManagedServices.csproj (hunk 3 above)
Comment (rajivnandivada) [reply src/SDKs/ManagedServices/Management.ManagedServices/Microsoft.Azure.Management.ManagedServices.csproj]: No I think it is a merge issue. I will add it back.

File: src/SDKs/_metadata/managedservices_resource-manager.txt (hunk 4 above)
Comment (rajivnandivada) [reply src/SDKs/_metadata/managedservices_resource-manager.txt]: will do thanks.

**Comment by rajivnandivada** [comment]
@dsgouda Thanks for your feedback. I addressed your feedback. Please review.

File: src/SDKs/ManagedServices/Management.ManagedServices/Properties/AssemblyInfo.cs [hunk 5]
```diff
@@ -1,19 +1,19 @@
...
 
-[assembly: AssemblyTitle("Microsoft Azure ManagedServices Management Library")]
-[assembly: AssemblyDescription("Provides Microsoft Azure ManagedServices management functions for managing the Microsoft Azure ManagedServices service.")]
-
-[assembly: AssemblyVersion("0.9.0")]
-[assembly: AssemblyFileVersion("0.9.0")]
+[assembly: AssemblyTitle("Microsoft Azure Managed Services Management Library")]
+[assembly: AssemblyDescription("Provides management functionality for Microsoft Azure Maanged Services.")]
 
+[assembly: AssemblyVersion("0.0.9.0")]
```
Comment (dsgouda) [inline src/SDKs/ManagedServices/Management.ManagedServices/Properties/AssemblyInfo.cs]: This should be bumped to `1.0.0.0` as well

File: src/SDKs/ManagedServices/Management.ManagedServices/Properties/AssemblyInfo.cs [hunk 6]
```diff
@@ -1,19 +1,19 @@
...
+[assembly: AssemblyFileVersion("0.0.9.0")]
 [assembly: AssemblyConfiguration("")]
 [assembly: AssemblyCompany("Microsoft")]
-[assembly: AssemblyProduct("Microsoft Azure .NET SDK")]
```
Comment (dsgouda) [inline src/SDKs/ManagedServices/Management.ManagedServices/Properties/AssemblyInfo.cs]: Please undo these changes

File: src/SDKs/ManagedServices/ManagedServices.Test/ManagedServices.Test.csproj [hunk 7]
```diff
@@ -0,0 +1,35 @@
+<Project Sdk="Microsoft.NET.Sdk" ToolsVersion="15.0">
+  <!-- Please do not move/edit code below this line -->
+  <Import Condition=" Exists('$([MSBuild]::GetPathOfFileAbove(AzSdk.test.reference.props))') " Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.test.reference.props'))" />
+  <!-- Please do not move/edit code below this line -->
+  <PropertyGroup>
+    <PackageId>Foo</PackageId>
```
Comment (dsgouda) [inline src/SDKs/ManagedServices/ManagedServices.Test/ManagedServices.Test.csproj]: Please fix the name and other tags
AssemblyVersion must be set to 1.0..0 and never changed for test projects

File: src/SDKs/ManagedServices/ManagedServices.Test/ManagedServices.Test.csproj [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    <VersionPrefix>1.0.0-preview</VersionPrefix>
+    <AssemblyName>Foo</AssemblyName>
+    <Description>BatchAI.Tests;</Description>
```
Comment (dsgouda) [inline src/SDKs/ManagedServices/ManagedServices.Test/ManagedServices.Test.csproj]: Please create an `AssemblyInfo.cs` file for the test project too and ensure the properties are appropriately set

File: src/SDKs/ManagedServices/ManagedServices.Test/ManagedServices.Test.csproj [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+  </PropertyGroup>
+  <PropertyGroup>
+    <TargetFrameworks>netcoreapp2.0</TargetFrameworks>
+  </PropertyGroup>
+  <ItemGroup>
+    <None Remove="SessionRecords\ManagedServices.Tests.ScenarioTests.RegistrationAssignmentTests\RegistrationAssignmentTests_CRUD.json" />
```
Comment (dsgouda) [inline src/SDKs/ManagedServices/ManagedServices.Test/ManagedServices.Test.csproj]: Is this remove intentional?

File: src/SDKs/ManagedServices/ManagedServices.sln [hunk 10]
```diff
@@ -1,31 +1,31 @@
...
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.Azure.Management.ManagedServices", "Management.ManagedServices\Microsoft.Azure.Management.ManagedServices.csproj", "{7909B3D2-7E61-4170-999A-49BC0A368DE1}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "ManagedServices.Test", "ManagedServices.Test\ManagedServices.Test.csproj", "{4AAA1DE9-6AAE-402B-A6E1-296EC46A0063}"
 EndProject
 Global
 	GlobalSection(SolutionConfigurationPlatforms) = preSolution
 		Debug|Any CPU = Debug|Any CPU
 		Release|Any CPU = Release|Any CPU
 	EndGlobalSection
 	GlobalSection(ProjectConfigurationPlatforms) = postSolution
-		{AD9682BB-EEE5-4CD1-AACB-4019FD9830D2}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
```
Comment (dsgouda) [inline src/SDKs/ManagedServices/ManagedServices.sln]: Undo changes if unintended

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: src/SDKs/ManagedServices/ManagedServices.Test/ManagedServices.Test.csproj (hunk 9 above)
Comment (rajivnandivada) [reply src/SDKs/ManagedServices/ManagedServices.Test/ManagedServices.Test.csproj]: No, it was not.

File: src/SDKs/ManagedServices/Management.ManagedServices/Properties/AssemblyInfo.cs (hunk 5 above)
Comment (rajivnandivada) [reply src/SDKs/ManagedServices/Management.ManagedServices/Properties/AssemblyInfo.cs]: Ok

File: src/SDKs/ManagedServices/Management.ManagedServices/Properties/AssemblyInfo.cs (hunk 6 above)
Comment (rajivnandivada) [reply src/SDKs/ManagedServices/Management.ManagedServices/Properties/AssemblyInfo.cs]: done

File: src/SDKs/ManagedServices/ManagedServices.Test/ManagedServices.Test.csproj (hunk 7 above)
Comment (rajivnandivada) [reply src/SDKs/ManagedServices/ManagedServices.Test/ManagedServices.Test.csproj]: Done

File: src/SDKs/ManagedServices/ManagedServices.Test/ManagedServices.Test.csproj (hunk 8 above)
Comment (rajivnandivada) [reply src/SDKs/ManagedServices/ManagedServices.Test/ManagedServices.Test.csproj]: ok will do

File: src/SDKs/ManagedServices/ManagedServices.sln (hunk 10 above)
Comment (rajivnandivada) [reply src/SDKs/ManagedServices/ManagedServices.sln]: It was not intentional.

File: src/SDKs/ManagedServices/ManagedServices.Test/ManagedServices.Test.csproj [hunk 11]
```diff
@@ -0,0 +1,40 @@
+<Project Sdk="Microsoft.NET.Sdk" ToolsVersion="15.0">
+  <!-- Please do not move/edit code below this line -->
+  <Import Condition=" Exists('$([MSBuild]::GetPathOfFileAbove(AzSdk.test.reference.props))') " Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.test.reference.props'))" />
+  <!-- Please do not move/edit code below this line -->
+  <PropertyGroup>
+    <PackageId>ManagedServices.Test</PackageId>
+    <VersionPrefix>1.0.0.0</VersionPrefix>
```
Comment (dsgouda) [inline src/SDKs/ManagedServices/ManagedServices.Test/ManagedServices.Test.csproj]: Please replace this line with `<Version>1.0.0</Version>`

File: src/SDKs/ManagedServices/ManagedServices.Test/ManagedServices.Test.csproj [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+    <AssemblyName>ManagedServices.Test</AssemblyName>
+    <Description>ManagedServices.Test class library</Description>
+  </PropertyGroup>
+  <PropertyGroup>
+    <TargetFrameworks>netcoreapp2.0</TargetFrameworks>
+  </PropertyGroup>
+  <ItemGroup>
```
Comment (dsgouda) [inline src/SDKs/ManagedServices/ManagedServices.Test/ManagedServices.Test.csproj]: Instead of picking individual json files, select all *.json files under the directory, this way new test recordings won't be missed

File: src/SDKs/ManagedServices/ManagedServices.Test/ManagedServices.Test.csproj [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+    <Content Include="SessionRecords\ManagedServices.Tests.ScenarioTests.RegistrationAssignmentTests\RegistrationAssignmentTests_CRUD.json">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </Content>
+  </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="..\Management.ManagedServices\Microsoft.Azure.Management.ManagedServices.csproj" />
+  </ItemGroup>
+  <ItemGroup>
+    <Reference Include="System">
```
Comment (dsgouda) [inline src/SDKs/ManagedServices/ManagedServices.Test/ManagedServices.Test.csproj]: Is this line intended?

**Review by dsgouda — CHANGES_REQUESTED** [review]
Looks great, just needs a few changes

File: src/SDKs/ManagedServices/ManagedServices.Test/ManagedServices.Test.csproj (hunk 11 above)
Comment (rajivnandivada) [reply src/SDKs/ManagedServices/ManagedServices.Test/ManagedServices.Test.csproj]: will do

File: src/SDKs/ManagedServices/ManagedServices.Test/ManagedServices.Test.csproj (hunk 12 above)
Comment (rajivnandivada) [reply src/SDKs/ManagedServices/ManagedServices.Test/ManagedServices.Test.csproj]: good idea, thanks

**Comment by dsgouda** [comment]
@rajivnandivada Please join the Azure github org [here](https://aka.ms/azuregithub)

File: src/SDKs/ManagedServices/ManagedServices.Test/ManagedServices.Test.csproj (hunk 13 above)
Comment (rajivnandivada) [reply src/SDKs/ManagedServices/ManagedServices.Test/ManagedServices.Test.csproj]: No. Copy/paste error. I removed it.

**Comment by rajivnandivada** [comment]
@dsgouda I am a member of the Azure github

image

**Comment by rajivnandivada** [comment]
The related Swagger spec is now merged.

**Review by dsgouda — CHANGES_REQUESTED** [review]
Please regenerate the code and update the pr

**Review by dsgouda — APPROVED** [review]
LGTM
